pkg/kubernetes/nodes.go

```diff
@@ -96,3 +108,266 @@ func (k *Kubernetes) NodesTop(ctx context.Context, options NodesTopOptions) (*me
 	convertedMetrics := &metrics.NodeMetricsList{}
 	return convertedMetrics, metricsv1beta1api.Convert_v1beta1_NodeMetricsList_To_metrics_NodeMetricsList(versionedMetrics, convertedMetrics, nil)
 }
+
+// NodeFilesOptions contains options for node file operations
+type NodeFilesOptions struct {
+	NodeName   string
+	Operation  string // "put", "get", "list"
+	SourcePath string
+	DestPath   string
+	Namespace  string
+	Image      string
+	Privileged bool
+}
+
+// NodesFiles handles file operations on a node filesystem by creating a privileged pod
+func (k *Kubernetes) NodesFiles(ctx context.Context, opts NodeFilesOptions) (string, error) {
+	// Set defaults
+	if opts.Namespace == "" {
+		opts.Namespace = "default"
+	}
+	if opts.Image == "" {
+		opts.Image = "busybox"
+	}
```

Comment:
Since the purpose of the tool/feature is to perform file operations is there any value in being able to customize the used image?

I understand, that changing the image might also break the implementation. Wouldn't it be better to use a specific image that can be freely pulled and that worked on any cluster?